src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -88,7 +88,7 @@ public static class AttributeMatcher
             {
                 if (!context.AllAttributes.ContainsKey(attribute) ||
                     context.AllAttributes[attribute] == null ||
-                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute] as string))
+                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute].ToString()))
```

Comment:
Even if it is `Nullable<T>` we would still need this change. And we should not modify the mode to have no required attributes because we would end up executing the matching code-path even for simple script sources like `<script src="test.js"></script>`